com.unity.shadergraph/Documentation~/Refract-Node.md

```diff
@@ -0,0 +1,42 @@
+# Refract Node
+
+## Description
+
+Generates a refraction [Ibn Sahl-Snell-Descartes's Law](https://en.wikipedia.org/wiki/Snell%27s_law). Based on an **Incident** vector, the **Normal** of the surface and **Eta** we produce the refracted vector.
+
+Based on Ibn Sahl's Law, for a given refractive index of the medium, we can reach an angle where all the surface behave like a perfect mirror, that's why in **Safe** Mode generate a null vector, this implementation is here to not double the reflection intensity which is handle by Fresnel.
+
+## Ports
+
+| Name        | Direction           | Type  | Binding | Description |
+|:------------ |:-------------|:-----|:---|:---|
+| Incident      | Input | Vector | None | Incident normalied vector to the surface (From surface to source, for instance from pixel to eye). |
```

Comment:
I'm not sure what this is supposed to mean. Is this the vector that represents the direction of the light as it hits the surface and reflects back, or what?

Reply:
That represents what we want to be refracted, one use case would be light to pixel/surface, or eyes to surface/pixel.
But it can be used as the user wants for multi bounce inside a bottle for instance.

Reply:
> The normalized vector of what should be refracted to the surface causing the refraction. For example, this could be from a light source to a pixel or the surface, or from the eye to the pixel or surface. 

Please change the text to the above.